doc/dev/adapter-development-walkthrough.md

```diff
@@ -0,0 +1,824 @@
+function to allow operators to pass the adapter specific config and for the adapter to validate the operator provided
+config. Copy the following code and the bold text shows the new added code.
+
+<pre>
+package mysampleadapter
+
+import (
+    // "github.com/gogo/protobuf/types"
+	"context"
+	<b>"os"</b>
+	<b>"path/filepath"</b>
+
+	</b>"istio.io/mixer/adapter/mysampleadapter/config"</b>
+	"istio.io/mixer/pkg/adapter"
+    "istio.io/mixer/template/metric"
+)
+
+type (
+	builder struct {
+		<b>adpCfg *config.Params</b>
+	}
+	handler struct {
+		<b>f *os.File</b>
+	}
+)
+
+// ensure types implement the requisite interfaces
+var _ metric.HandlerBuilder = &builder{}
+var _ metric.Handler = &handler{}
+
+///////////////// Configuration-time Methods ///////////////
+
+// adapter.HandlerBuilder#Build
+func (b *builder) Build(ctx context.Context, env adapter.Env) (adapter.Handler, error) {
+	<b>file, err := os.Create(b.adpCfg.FilePath)
+	return &handler{f: file}, err</b>
+
+}
+
+// adapter.HandlerBuilder#SetAdapterConfig
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	<b>b.adpCfg = cfg.(*config.Params)</b>
+}
+
+// adapter.HandlerBuilder#Validate
+func (b *builder) Validate() (ce *adapter.ConfigErrors) {
+	<b>// Check if the path is valid
+	if _, err := filepath.Abs(b.adpCfg.FilePath); err != nil {
+		ce = ce.Append("file_path", err)
+	}
+	return</b>
+}
+
+// metric.HandlerBuilder#SetMetricTypes
+func (b *builder) SetMetricTypes(types map[string]*metric.Type) {
+}
+
+////////////////// Request-time Methods //////////////////////////
+// metric.Handler#HandleMetric
+func (h *handler) HandleMetric(ctx context.Context, insts []*metric.Instance) error {
+	return nil
+}
+
+// adapter.Handler#Close
+func (h *handler) Close() error {
+	<b>return h.f.Close()</b>
+}
+
+////////////////// Bootstrap //////////////////////////
+// GetInfo returns the adapter.Info specific to this adapter.
+func GetInfo() adapter.Info {
+	return adapter.Info{
+		Name:        "mysampleadapter",
+		Description: "Logs the metric calls into a file",
+		SupportedTemplates: []string{
+			metric.TemplateName,
+		},
+		NewBuilder:    func() adapter.HandlerBuilder { return &builder{} },
+		<b>DefaultConfig: &config.Params{},</b>
+	}
+}
+</pre>
+
+
+Just to ensure everything is good, let's build the code
+
+```
+bazel build ...
+```
+
+
+The build output on the terminal look like
+
+*INFO: Found 1 target...*
+
+*Target //adapter/*mysampleadapter*:go_default_library up-to-date:*
+
+*bazel-bin/adapter/*mysampleadapter*/~lib~/istio.io/mixer/adapter/*mysampleadapter*.a*
+
+# Step 4: Write business logic into your adapter.
+
+Print Instance and associated Type information in the file configured via adapter config. This requires storing the
+metric type information during configure-time and using it during request-time. To add this functionality into
+'mysampleadapter.go'`, `copy the following code and the bold text shows the newly added code.
```

Comment:
Here and elsewhere, you can just say:

"Update file XXX to look like the following"